lasagne/layers/recurrent.py

```diff
@@ -0,0 +1,1353 @@
+        The layer feeding into this layer, or the expected input shape.
+    num_units : int
+        Number of hidden units in the layer.
+    W_in_to_hid : Theano shared variable, numpy array or callable
+        Initializer for input-to-hidden weight matrix (:math:`W_x`).
+    W_hid_to_hid : Theano shared variable, numpy array or callable
+        Initializer for hidden-to-hidden weight matrix (:math:`W_h`).
+    b : Theano shared variable, numpy array, callable or None
+        Initializer for bias vector (:math:`b`). If None is provided there will
+        be no bias.
+    nonlinearity : callable or None
+        Nonlinearity to apply when computing new state (:math:`\sigma`). If
+        None is provided, no nonlinearity will be applied.
+    hid_init : callable, np.ndarray, theano.shared or TensorVariable
+        Passing in a TensorVariable allows the user to specify
+        the value of `hid_init` (:math:`h_0`). In this mode `learn_init` is
+        ignored.
+    backwards : bool
+        If True, process the sequence backwards and then reverse the
+        output again such that the output from the layer is always
+        from :math:`x_1` to :math:`x_n`.
+    learn_init : bool
+        If True, initial hidden values are learned. If `hid_init` is a
+        TensorVariable then `learn_init` is ignored.
+    gradient_steps : int
+        Number of timesteps to include in the backpropagated gradient.
+        If -1, backpropagate through the entire sequence.
+    grad_clipping : False or float
+        If a float is provided, the gradient messages are clipped during the
+        backward pass.  If False, the gradients will not be clipped.  See [1]_
+        (p. 6) for further explanation.
+    unroll_scan : bool
+        If True the recursion is unrolled instead of using scan. For some
+        graphs this gives a significant speed up but it might also consume
+        more memory. When `unroll_scan` is True, backpropagation always
+        includes the full sequence, so `gradient_steps` must be set to -1 and
+        the input sequence length must be known at compile time (i.e., cannot
+        be given as None).
+    precompute_input : bool
+        If True, precompute input_to_hid before iterating through
+        the sequence. This can result in a speedup at the expense of
+        an increase in memory usage.
+    mask_input : :class:`lasagne.layers.Layer`
+        Layer which allows for a sequence mask to be input, for when sequences
+        are of variable length.  Default `None`, which means no mask will be
+        supplied (i.e. all sequences are of the same length).
+
+    References
+    ----------
+    .. [1] Graves, Alex: "Generating sequences with recurrent neural networks."
+           arXiv preprint arXiv:1308.0850 (2013).
+    """
+    def __init__(self, incoming, num_units,
+                 W_in_to_hid=init.Uniform(),
+                 W_hid_to_hid=init.Uniform(),
```

Comment:
No, there isn't really consensus for how to initialize RNN parameters in the way that there are for feedforward networks.  Uniform seemed like a sensible least-common denominator default.  The LSTM and GRU layers use .1-std normals, because I've seen that in a few other libraries.  I can unify it all if you want.  If you or anyone has input on sensible defaults, that would be cool too.